### ISSUE TITLE
could not find compile_error in $crate on nom 4.2

### ISSUE THREAD

**Issue by Veykril** [Geal/nom#862]
It looks like 4.2 broke the `compile_error!` usage. When switching to 4.2 whenever a `compile_error!` gets triggered for me the compiler only emits a  `error[E0433]: failed to resolve: could not find compile_error in $crate` instead of the expected helpful message. On `~4.1` the expected `error: do_parse uses >> as separator, not ~` gets shown instead.

## Prerequisites

Here are a few things you should provide to help me understand the issue:

- Rust version : rustc 1.33.0-nightly (c76f3c374 2019-01-18)
- nom version : 4.2.0
- nom compilation features used:

## Test case

```
#[macro_use]
extern crate nom;

use nom::IResult;

pub struct JmxBMesh {}

impl JmxBMesh {
    pub fn new(input: &[u8]) -> IResult<&[u8], Self> {
        do_parse!(input,
            tag!("JMXVBMS 0110") ~
            ( JmxBMesh {} )
        )
    }
}
```

**Comment by m-cat** [comment]
I'm also seeing this. Version 4.2

**Comment by Geal** [comment]
right, `compile_error` is not a local macro, I'll fix that and release a new version

**Comment by Geal** [comment]
this should be fixed by 34df620 (new version is not released yet)

**Comment by Geal** [comment]
this is fixed in nom 4.2.2
